static/js/components/tiles/chart_action_icons.tsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Icons for downloading, linking, embedding, and ranking in chart footer.
+ */
+
+import React, { useEffect, useRef, useState } from "react";
+import { UncontrolledTooltip } from "reactstrap";
+
+import {
+  GA_EVENT_TILE_DOWNLOAD,
+  GA_EVENT_TILE_EXPLORE_MORE,
+  GA_PARAM_TILE_TYPE,
+  triggerGAEvent,
+} from "../../shared/ga_events";
+import { ChartDownload, ChartDownloadSpec } from "./modal/chart_download";
+
+/** Component for a single action icon */
+
+interface ActionIconPropType {
+  // Container element the tooltip should attach to
+  container?: HTMLElement;
```

Comment:
is it necessary to set the container in the tooltips? maybe it is- something to do with tooltip rendering in a web component?